cdap-app-templates/cdap-etl/hydrator-spark-core/src/main/java/co/cask/cdap/etl/spark/function/PluginFunctionContext.java

```diff
@@ -72,10 +74,12 @@ public PluginFunctionContext(String stageName, JavaSparkExecutionContext sec, Pi
     }
     this.arguments = arguments;
     this.pipelineStr = GSON.toJson(pipelinePhase);
+    this.sec = sec;
+    this.namespace = sec.getNamespace();
   }
 
   public <T> T createPlugin() throws Exception {
-    MacroEvaluator macroEvaluator = new DefaultMacroEvaluator(arguments, logicalStartTime);
+    MacroEvaluator macroEvaluator = new DefaultMacroEvaluator(arguments, logicalStartTime, sec, namespace);
```

Comment:
it looks like the latest develop has a getSecureStore() method in JavaSparkExecutionContext.  You need to use that method and store just the SecureStore as a field of this class, otherwise Spark serialization/deserialization won't work.